src/replica/bulk_load/replica_bulk_loader.cpp

```diff
@@ -403,80 +387,112 @@ error_code replica_bulk_loader::download_sst_files(const std::string &remote_dir
         return ERR_FILE_OPERATION_FAILED;
     }
 
+    // start download
+    _is_downloading.store(true);
+    _download_task = tasking::enqueue(
+        LPC_BACKGROUND_BULK_LOAD,
+        tracker(),
+        std::bind(
+            &replica_bulk_loader::download_files, this, provider_name, remote_dir, local_dir));
+    return ERR_OK;
+}
+
+// ThreadPool: THREAD_POOL_DEFAULT
+void replica_bulk_loader::download_files(const std::string &provider_name,
+                                         const std::string &remote_dir,
+                                         const std::string &local_dir)
+{
+    FAIL_POINT_INJECT_F("replica_bulk_loader_download_files", [](string_view) {});
+
+    ddebug_replica("start to download files");
     dist::block_service::block_filesystem *fs =
         _stub->_block_service_manager.get_or_create_block_filesystem(provider_name);
 
+    zauto_write_lock l(_lock);
```

Comment:
Could you tell us the protect scope of `_lock`?

Reply:
Okay~ The related code has been updated.